ibis/backends/pyspark/compiler.py

```diff
@@ -1,2137 +1,272 @@
+#     > set, you can use some large number for the second parameter. This statement
+#     > retrieves all rows from the 96th row to the last:
+#     >
+#     > SELECT * FROM tbl LIMIT 95,18446744073709551615;
+#     """
+#     if _.n is None and _.offset is not None:
+#         some_large_number = (1 << 64) - 1
+#         return _.copy(n=some_large_number)
+#     return _
+
+
+@public
+class PySparkCompiler(SQLGlotCompiler):
+    __slots__ = ()
+
+    dialect = "spark"
+    type_mapper = PySparkType
+    # rewrites = (
+    #     rewrite_limit,
+    #     rewrite_sample,
+    #     rewrite_first_to_first_value,
+    #     rewrite_last_to_last_value,
+    #     exclude_unsupported_window_frame_from_ops,
+    #     exclude_unsupported_window_frame_from_row_number,
+    #     rewrite_empty_order_by_window,
+    #     *SQLGlotCompiler.rewrites,
+    # )
+    quoted = True
+
+    NAN = sge.Literal.number("FLOAT('NaN')")
+    POS_INF = sge.Literal.number("FLOAT('Infinity')")
+    NEG_INF = sge.Literal.number("FLOAT('-Infinity')")
+
+    def _aggregate(self, funcname: str, *args, where):
+        func = self.f[funcname]
+        if where is not None:
+            args = tuple(self.if_(where, arg, NULL) for arg in args)
+        return func(*args)
+
+    # @staticmethod
+    # def _generate_groups(groups):
+    #     return groups
+
+    @singledispatchmethod
+    def visit_node(self, op, **kwargs):
+        return super().visit_node(op, **kwargs)
+
+    def visit_NonNullLiteral(self, op, *, value, dtype):
+        if dtype.is_uuid():
+            return sge.Literal.string(value)
         else:
-            raise NotImplementedError(
-                f"Unrecognized type in selections: {type(selection)}"
-            )
-    if col_in_selection_order:
-        result_table = result_table[col_in_selection_order]
-
-    if col_to_drop:
-        result_table = result_table.drop(*col_to_drop)
-
-    if op.sort_keys:
-        sort_cols = [
-            t.translate(key, scope=scope, timecontext=timecontext, **kwargs)
-            for key in op.sort_keys
-        ]
-        result_table = result_table.sort(*sort_cols)
-
-    return filter_by_time_context(result_table, timecontext, adjusted_timecontext)
+            return None
 
-
-@compiles(ops.SortKey)
-def compile_sort_desc(t, op, **kwargs):
-    col = t.translate(op.expr, **kwargs)
-    return col.asc() if op.ascending else col.desc()
-
-
-def compile_nan_as_null(compile_func):
-    @functools.wraps(compile_func)
-    def wrapper(t, op, *args, **kwargs):
-        compiled = compile_func(t, op, *args, **kwargs)
-        if options.pyspark.treat_nan_as_null and isinstance(op.dtype, dt.Floating):
-            return F.nanvl(compiled, F.lit(None))
+    @visit_node.register(ops.Field)
+    def visit_Field(self, op, **kwargs):
+        result = super().visit_node(op, **kwargs)
+        if op.dtype.is_timestamp():
+            return self.cast(result, dt.timestamp)
```

Comment:
guess this is pre-existing? why do we need to cast?

Reply:
Not needed, removing.